pkg/netceptor/tlsconfig.go

```diff
@@ -65,7 +79,16 @@ func (cfg TLSServerCfg) Prepare() error {
 		tlscfg.ClientAuth = tls.NoClientCert
 	}
 
-	return MainInstance.SetServerTLSConfig(cfg.Name, tlscfg)
+	serverConf := tlscfg
+	if cfg.VerifyClientNodeID {
+		tlscfg.NextProtos = []string{"netceptor"}
+		serverConf = &tls.Config{}
+		serverConf.GetConfigForClient = func(hi *tls.ClientHelloInfo) (*tls.Config, error) {
+			tlscfg.VerifyPeerCertificate = getClientValidator(hi, tlscfg.ClientCAs)
```

Comment:
This seems to be modifying the outer-scope tlscfg?

Reply:
yeah it does, and I am not to sure how to get around it.